contributors/design-proposals/scheduling/scheduling-framework.md

```diff
@@ -1,436 +1,620 @@
-cycles (end of 2018). If things go well, we will start offering it as an
-alternative to the scheduler v1 by the end of Q1 2019 and start the deprecation
-of scheduler v1. We will make it the default scheduler of Kubernetes in Q2 2019,
-but we will keep the option of using scheduler v1 for at least two more release
-cycles.
-</s>
+## Plugin Lifecycle
+
+### Initialization
+
+There are two steps to plugin initialization. First,
+[plugins are registered](#plugin-registration). Second, the scheduler uses its
+configuration to decide which plugins to instantiate. If a plugin registers for
+multiple extension points, *it is instantiated only once*.
+
+When a plugin is instantiated, it is passed [config args](#optional-args) and a
+[`PluginHandle`](#pluginhandle).
+
+### Concurrency
+
+There are two types of concurrency that plugin writers should consider. A plugin
+might be invoked several times concurrently when evaluating multiple nodes, and
+a plugin may be called concurrently from *different
+[scheduling cycles](#scheduling-cycle)*.
+
+In the main thread of the scheduler, only one scheduling cycle is processed at a
+time. Any extension point up to and including [reserve](#reserve) will be
+finished before the next scheduling cycle begins*. After the reserve phase, the
+[permit](#permit) and [bind](#bind) phases are executed asynchronously. This
+means that a plugin could be called concurrently from two different scheduling
+cycles, provided that at least one of the calls is to an extension point after
+reserve. Stateful plugins should take care to handle these situations.
+
+Finally, [un-reserve](#un-reserve) plugins may be called from either the Permit
+thread or the Bind thread, depending on how the pod was rejected.
+
+\* *The queue sort extension point is a special case. It is not part of a
+scheduling cycle and may be called concurrently for many pod pairs.*
+
+![image](images/scheduling-framework-threads.png)
+
+## Configuring Plugins
+
+The scheduler's component configuration will allow for plugins to be enabled,
+disabled, or otherwise configured. Plugin configuration is separated into two
+parts.
+
+1.  A list of enabled plugins for each extension point (and the order they
+    should run in)
+1.  An optional set of custom plugin arguments for each plugin
+
+The plugin configuration is organized by extension points. A plugin that
+registers with multiple points must be included in each list.
+
+```go
+type KubeSchedulerConfiguration struct {
+    // ... other fields
+    Plugins      Plugins
+    PluginConfig []PluginConfig
+}
+
+type Plugins struct {
+    QueueSort      []Plugin
+    PreFilter      []Plugin
+    Filter         []Plugin
+    PostFilter     []Plugin
+    Score          []Plugin
+    NormalizeScore []Plugin
+    Reserve        []Plugin
+    Permit         []Plugin
+    PreBind        []Plugin
+    Bind           []Plugin
+    PostBind       []Plugin
+    UnReserve      []Plugin
```

Comment:
i believe not everything needs to be specified.

Reply:
Yeah. In this design, omitting any details from this section results in the default behavior (similar to default algorithm provider today). I've added some clarification.